### PR TITLE
fix plus tests

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Issue : we are not handling the case where update1 or below vs is present and parallel is selected.
added unit tests and manual testing

### CODE REVIEW

File: Tasks/VsTest/task.json [hunk 1]
```diff
@@ -1,106 +1,106 @@
...
+    "category": "Test",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 1,
+        "Minor": 0,
+        "Patch": 41
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: version?

File: Tasks/VsTest/vstest.ts [hunk 2]
```diff
@@ -108,8 +108,13 @@ function invokeVSTest(testResultsDirectory: string): Q.Promise<number> {
                     setRunInParallellIfApplicable(vsVersion);
                     setupRunSettingsFileForParallel(runInParallel, overriddenSettingsFile)
                         .then(function(parallelRunSettingsFile) {
-                            var vsCommon = tl.getVariable("VS" + vsVersion + "0COMNTools");
-                            var vstestLocation = path.join(vsCommon, "..\\IDE\\CommonExtensions\\Microsoft\\TestWindow\\vstest.console.exe");
+                            var vstestPath = tl.getVariable("VSTest_" + vsVersion.toFixed(1));
+                            if (!vstestPath) {
+                                tl.warning(tl.loc('VstestNotFound', vsVersion));
+                                defer.resolve(1);
```
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: why 1

File: Tasks/VsTest/vstest.ts [hunk 3]
```diff
@@ -432,16 +442,17 @@ function setRunInParallellIfApplicable(vsVersion: number) {
     if (runInParallel) {
         if (!isNaN(vsVersion) && vsVersion >= 14) {
             var vs14Common = tl.getVariable("VS140COMNTools");
-            if ((vs14Common && pathExistsAsFile(path.join(vs14Common, "..\\IDE\\CommonExtensions\\Microsoft\\TestWindow\\TE.TestModes.dll"))) || vsVersion > 14) {
+            if (vsVersion > 14 || (vs14Common && pathExistsAsFile(path.join(vs14Common, "..\\IDE\\CommonExtensions\\Microsoft\\TestWindow\\TE.TestModes.dll")))) {
```
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: what is temodes.dll?

Comment (AshwiniChalla) [inline Tasks/VsTest/vstest.ts]: this dll determines if it is below update 1

File: Tasks/VsTest/vstest.ts (hunk 2 above)
Comment (AshwiniChalla) [inline Tasks/VsTest/vstest.ts]: 1 means task fails

File: Tasks/VsTest/task.json (hunk 1 above)
Comment (AshwiniChalla) [inline Tasks/VsTest/task.json]: will change

File: Tasks/VsTest/vstest.ts (hunk 3 above)
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: and we r sure it wont be removed going forward? we cant guarantee that correct?

---
In reply to: [67283885](https://github.com/Microsoft/vsts-tasks/pull/1886#discussion_r67283885) [](ancestors = 67283885)

File: Tasks/VsTest/vstest.ts (hunk 2 above)
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: then the warning shuld be made error?

---
In reply to: [67283921](https://github.com/Microsoft/vsts-tasks/pull/1886#discussion_r67283921) [](ancestors = 67283921)

**Comment by acesiddhu** [comment]
:shipit:

**Pull request merged**